cocotb/scheduler.py

```diff
@@ -769,7 +769,7 @@ def schedule(self, coroutine, trigger=None):
             self.unschedule(coroutine)
 
         # Don't handle the result if we're shutting down
-        if self._terminate:
+        if self._terminate or self._test._outcome is not None:
             return
```

Comment:
I think this needs a comment explaining why we check `_outcome`. Perhaps 

```suggestion
        if self._terminate:
            return
        # the test has completed before termination began. This can happen if XXX.
        if self._test._outcome is not None:
            return
```